src/EnumDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MabeEnumPHPStan;
+
+use MabeEnum\Enum;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+
+class EnumDynamicReturnTypeExtension implements DynamicMethodReturnTypeExtension
```

Comment:
I would name it somehow with "values" specified. We can later add support for `is` or other public methods and it will probably not be part of this class.

Reply:
I would keep it this way and if we add more this can still act as extension entry point but using different classes internally.

Reply:
It is probably the same thing what PHPStan does internally, but why not.